WooCommerce/WooCommerceTests/ViewRelated/Survey/SurveyViewControllerTests.swift

```diff
@@ -57,7 +57,30 @@ final class SurveyViewControllerTests: XCTestCase {
         let mirror = try self.mirror(of: viewController)
 
         // Fakes a form submission GET navigation
-        let navigationAction = FormSubmittedNavigationAction(httpMethod: "GET")
+        let navigationAction = FormSubmittedNavigationAction.nonCompletedAction
+        viewController.webView(mirror.webView, decidePolicyFor: navigationAction, decisionHandler: { _ in })
+        waitForExpectations(timeout: Constants.expectationTimeout)
+
+        // Then
+        XCTAssertFalse(surveyCompleted)
+    }
+
+    func test_it_does_not_complete_after_receiving_a_form_submitted_empty_callback_request() throws {
```

Comment:
⚠️ This test and the test above takes at least 10 seconds. A second is quite expensive in CircleCI, especially when we have more developers submitting PRs in the future. 😅 

I did a quick test and it looks like we can use the `decisionHandler` for waiting. 

```swift
    func test_it_does_not_complete_after_receiving_a_form_submitted_empty_callback_request() throws {
        // Given
        var surveyCompleted = false
        let viewController = SurveyViewController(survey: .inAppFeedback, onCompletion: {
            surveyCompleted = true
        })

        // When
        _ = try XCTUnwrap(viewController.view)
        let mirror = try self.mirror(of: viewController)

        // Fakes a form submission GET navigation
        let navigationAction = FormSubmittedNavigationAction.emptyAction

        waitForExpectation { exp in
            viewController.webView(mirror.webView, decidePolicyFor: navigationAction, decisionHandler: { _ in
                exp.fulfill()
            })
        }

        // Then
        XCTAssertFalse(surveyCompleted)
    }
```

This pattern seems to work fine when I used it for `test_it_completes_after_receiving_a_form_submitted_completed_callback_request` too so I think it's testing the functionality accurately. `:nervouslaugh:`

Also, thank you very much for testing this. This is a very important logic! 🙏

Reply:
Uuu nice idea! I did not like that waiting at all!